File: backend/src/main/java/com/supermarket/backend/entity/InventoryTransaction.java
package com.supermarket.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "inventory_transactions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class InventoryTransaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "transaction_number")
    private Integer transactionNumber;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_number", referencedColumnName = "product_number")
    private Product product;

    @Column(name = "stock_in_detail_number")
    private Integer stockInDetailNumber;

    @Column(name = "type")
    private String type;

    @Column(name = "quantity")
    private BigDecimal quantity;

    @Column(name = "reference_type")
    private String referenceType;

    @Column(name = "reference_id")
    private Integer referenceId;

    @Column(name = "reason")
    private String reason;

    @Column(name = "created_at")
    private LocalDateTime createdAt;
}

File: frontend/lib/models/inventory_transaction.dart
class InventoryTransaction {
  final int? transactionNumber;
  final int? productNumber;
  final String? productName;
  final String? type;
  final num? quantity;
  final String? unitName;
  final String? referenceType;
  final int? referenceId;
  final String? reason;
  final String? createdBy;
  final DateTime? createdAt;

  InventoryTransaction({
    this.transactionNumber,
    this.productNumber,
    this.productName,
    this.type,
    this.quantity,
    this.unitName,
    this.referenceType,
    this.referenceId,
    this.reason,
    this.createdBy,
    this.createdAt,
  });

  factory InventoryTransaction.fromJson(Map<String, dynamic> json) {
    return InventoryTransaction(
      transactionNumber: json['transactionNumber'],
      productNumber: json['productNumber'],
      productName: json['productName'],
      type: json['type'],
      quantity: json['quantity'],
      unitName: json['unitName'],
      referenceType: json['referenceType'],
      referenceId: json['referenceId'],
      reason: json['reason'],
      createdBy: json['createdBy'],
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
    );
  }
}
